prow/pod-utils/decorate/podspec.go

```diff
@@ -669,37 +677,49 @@ func decorate(spec *coreapi.PodSpec, pj *prowapi.ProwJob, rawEnv map[string]stri
 		*initUpload,
 		PlaceEntrypoint(pj.Spec.DecorationConfig, toolsMount),
 	)
-	spec.Containers[0].Env = append(spec.Containers[0].Env, KubeEnv(rawEnv)...)
+	for i, container := range spec.Containers {
+		spec.Containers[i].Env = append(container.Env, KubeEnv(rawEnv)...)
+	}
 
-	const ( // these values may change when/if we support multiple containers
-		prefix   = "" // unique per container
+	const (
 		previous = ""
 		exitZero = false
 	)
-	wrapperOptions, err := InjectEntrypoint(&spec.Containers[0], pj.Spec.DecorationConfig.Timeout.Get(), pj.Spec.DecorationConfig.GracePeriod.Get(), prefix, previous, exitZero, logMount, toolsMount)
-	if err != nil {
-		return fmt.Errorf("wrap container: %v", err)
+	var wrappers []wrapper.Options
+
+	for i, container := range spec.Containers {
+		prefix := container.Name
+		if len(spec.Containers) == 1 {
+			prefix = ""
+		}
+		wrapperOptions, err := InjectEntrypoint(&spec.Containers[i], pj.Spec.DecorationConfig.Timeout.Get(), pj.Spec.DecorationConfig.GracePeriod.Get(), prefix, previous, exitZero, logMount, toolsMount)
+		if err != nil {
+			return fmt.Errorf("wrap container: %v", err)
+		}
+		wrappers = append(wrappers, *wrapperOptions)
 	}
 
-	sidecar, err := Sidecar(pj.Spec.DecorationConfig, blobStorageOptions, blobStorageMounts, logMount, outputMount, encodedJobSpec, !RequirePassingEntries, *wrapperOptions)
+	sidecar, err := Sidecar(pj.Spec.DecorationConfig, blobStorageOptions, blobStorageMounts, logMount, outputMount, encodedJobSpec, !RequirePassingEntries, wrappers...)
 	if err != nil {
 		return fmt.Errorf("create sidecar: %v", err)
 	}
 
-	spec.Containers = append(spec.Containers, *sidecar)
-
 	spec.Volumes = append(spec.Volumes, logVolume, toolsVolume)
 	spec.Volumes = append(spec.Volumes, blobStorageVolumes...)
 	if outputVolume != nil {
 		spec.Volumes = append(spec.Volumes, *outputVolume)
 	}
 
 	if len(refs) > 0 {
-		spec.Containers[0].WorkingDir = DetermineWorkDir(codeMount.MountPath, refs)
-		spec.Containers[0].VolumeMounts = append(spec.Containers[0].VolumeMounts, codeMount)
+		for i, container := range spec.Containers {
+			spec.Containers[i].WorkingDir = DetermineWorkDir(codeMount.MountPath, refs)
```

Comment:
You already have access to `container`.

Reply:
I think this is correct as is. `container` is a copy of `spec.Containers[i]` which is a struct rather than a reference in this case so changes to `container` are not reflected back to the slice.